setuptools/tests/config/test_expand.py

```diff
@@ -0,0 +1,106 @@
+import os
+
+import pytest
+
+from distutils.errors import DistutilsOptionError
+from setuptools.config import expand
+from setuptools.sandbox import pushd
+
+
+def write_files(files, root_dir):
+    for file, content in files.items():
+        path = root_dir / file
+        path.parent.mkdir(exist_ok=True, parents=True)
+        path.write_text(content)
+
+
+def test_glob_relative(tmp_path):
+    files = {
+        "dir1/dir2/dir3/file1.txt",
+        "dir1/dir2/file2.txt",
+        "dir1/file3.txt",
+        "a.ini",
+        "b.ini",
+        "dir1/c.ini",
+        "dir1/dir2/a.ini",
+    }
+
+    write_files({k: "" for k in files}, tmp_path)
+    patterns = ["**/*.txt", "[ab].*", "**/[ac].ini"]
+    with pushd(tmp_path):
+        assert set(expand.glob_relative(patterns)) == files
+    # Make sure the same APIs work outside cwd
+    assert set(expand.glob_relative(patterns, tmp_path)) == files
+
+
+def test_read_files(tmp_path):
+    files = {
+        "a.txt": "a",
+        "dir1/b.txt": "b",
+        "dir1/dir2/c.txt": "c"
+    }
+    write_files(files, tmp_path)
+    with pushd(tmp_path):
+        assert expand.read_files(list(files)) == "a\nb\nc"
+
+    with pushd(tmp_path / "dir1"), pytest.raises(DistutilsOptionError):
```

Comment:
It is highly recommended to always use a `match` argument together with `pytest.raises`/`pytest.warns`/`pytest.deprecated` so that the test is precise and doesn't pass based on side-effects.

Reply:
Thank you very much for the suggestion, I tried to add a match for all checks in this test file.